### ISSUE TITLE
Link to class enum in namespace not working in file reference

### ISSUE THREAD

**Issue by louwers** [doxygen/doxygen#9850]
**Describe the bug**
I have a function that returns an enum value. Documentation is generated for the enum but I cannot click on the return value to go to that documentation in a file reference (it works in a namespace reference). 

**Expected behavior**
I expect to be able to link to an class enum inside a namespace.

**Screenshots**
Here is a screenshot of documentation of a file without a link:

![image](https://user-images.githubusercontent.com/649392/218257568-095e22d8-45ba-437e-a8e8-30648c126c98.png)

And here is namespace documentation, where there is a link:

![image](https://user-images.githubusercontent.com/649392/218257583-c1514767-be0a-4ff2-84db-01a9ced8ae79.png)

**To Reproduce**

status.hpp

```cpp
/// @file

/// my namespace
namespace bla {

/// some enum
enum class Status { Done, Todo };

} // namespace bla
```

test.hpp

```cpp
/// @file

#include "status.hpp"

namespace bla {

Status someFunction();

} // namespace bla
```

[reproduction.zip](https://github.com/doxygen/doxygen/files/10713628/reproduction.zip)

**Version**
1.9.6 

**Stack trace**
N/A

**Additional context**
I asked about this on Stack Overflow as well: https://stackoverflow.com/questions/75408557/how-to-get-links-to-namespaced-enums-to-work-with-doxygen?noredirect=1#comment133066739_75408557

**Comment by doxygen** [comment]
Please verify if the referenced commit fixes the problem. Do not close the issue, as this will be done automatically when the next official version is released.

**Comment by louwers** [comment]
@doxygen Works now.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
